hw/hal/include/hal/hal_system.h

```diff
@@ -52,6 +54,17 @@ void hal_system_restart(void *img_start) __attribute((noreturn));
  */
 int hal_debugger_connected(void);
 
+#if MYNEWT_VAL(HAL_BREAK_HOOK)
+/* User defined function to be called before code is stopped in debugger */
+void hal_break_hook(void);
+#else
+#define hal_break_hook()
+#endif
+
+#ifndef HAL_DEBUG_BREAK
+#define HAL_DEBUG_BREAK() do { if (hal_debugger_connected()) { hal_break_hook(); hal_debug_break(); }} while (0)
+#endif
```

Comment:
Not that there is anything wrong with this but I do not like this "syntax". Very hard to read. Why is this just not a function? Something else that I might have missed in the other commits: is there any expectation that HAL_DEBUG_BREAK is defined with something else?

Reply:
@wes3  similar pattern was use in some files, but I'm not going to force this. I could drop () for once, or change it to static inline.
Macro with ifdef makes is easier to completely drop it if someone wants to like -DHAL_DEBUG_BREAK or such.